autoload/youcompleteme.vim

```diff
@@ -1357,7 +1357,7 @@ function! s:Complete()
   endif
   if len( s:completion.completions )
     let old_completeopt = &completeopt
-    set completeopt+=noselect
+    setlocal completeopt+=noselect
     call complete( s:completion.completion_start_column,
                  \ s:completion.completions )
     let &completeopt = old_completeopt
```

Comment:
Does this set the buffer local or global value? If it was previously globally set does this make it buffer local? I'm not sure that's desirable

Reply:
This does set the buffer-local value.If it was previously globally set, this does make it buffer-local.

I've manually tested this scenario:

1. Do some completion thing, so that we mess with buffer-local `completeopt`.
2. Manually mess with the global value. At the point, the buffer-local `completeopt` is destroyed, because global config changed.
3. Do more completion.

That worked as expected and the end result is that both global and buffer-local `completeopt` have the same value.

Can you think of more potentially broken scenarios to test?

Reply:
Nope. SGTM.